### PR TITLE
Add PHP Code coverage report as a GitHub action

### PR DESCRIPTION
**Pull request by mikkamp**
### Changes proposed in this Pull Request:

Resolves part of #864 

This PR adds a code coverage report as a PR comment. Since we run multiple unit tests in GitHub actions the coverage report is only generated once when the tests are run with the latest WP version and for PHP 8.0.

I've added the coverage using a flag `php-unit-tests`, the intention is to also include the coverage report for the JS unit tests, but that will be part of a followup PR.

> **Note**
There is currently no report stored for the develop branch. So it's only producing a coverage report of the current PR and isn't able to compare the difference between the two. Once this PR is merged to develop it will generate the report for develop which will allow the comparison to appear in subsequent PR's.

### Detailed test instructions:
1. View the code coverage report here in the [PR comment](https://github.com/woocommerce/google-listings-and-ads/pull/1915#issuecomment-1483222517) and confirm it contains the right test results (no comparison data available)
2. Confirm that in [the workflow runs](https://github.com/woocommerce/google-listings-and-ads/actions/runs/4532661162/jobs/7984348011?pr=1915) the coverage report is only generated for PHP 8.0 and WP latest, we should see skipped tasks that differ:
image

### Changelog entry
* Dev - Add PHP Code coverage report as GitHub action.

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -17,4 +17,11 @@
 			<directory suffix=".php">./tests/integration</directory>
 		</testsuite>
 	</testsuites>
+    <coverage includeUncoveredFiles="true">
```
Comment (puntope) [inline phpunit.xml.dist]: 💅  [Seems like this is the default value](https://docs.phpunit.de/en/9.5/code-coverage-analysis.html#:~:text=includeUncoveredFiles%3D%22true%22%20(default)%20means%20that%20all%20files%20are%20included%20in%20the%20code%20coverage%20report%20even%20if%20not%20a%20single%20line%20of%20code%20of%20such%20a%20file%20is%20executed), so no need for adding this .

File: .codecov.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    - 'tests/*'
+    - 'vendor/*'
+    - 'node_modules/*'
+
+comment:
+    layout: 'reach, diff, flags, files'
+    behavior: default
+    require_changes: false
+    require_base: no
+    require_head: yes
```
Comment (puntope) [inline .codecov.yml]: 💅 This is default to yes. So we can consider omit it.

File: .codecov.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ignore:
+    - 'bin/*'
+    - 'tests/*'
+    - 'vendor/*'
+    - 'node_modules/*'
+
+comment:
+    layout: 'reach, diff, flags, files'
+    behavior: default
+    require_changes: false
```
Comment (puntope) [inline .codecov.yml]: These three seems for me like default options

File: .codecov.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    require_base: no
+    require_head: yes
+    show_carryforward_flags: false
```
Comment (puntope) [inline .codecov.yml]: This I think is also false by default

File: .codecov.yml (hunk 3 above)
Comment (mikkamp) [reply .codecov.yml]: I thought so too, but apparently the default looked different without adding these settings. For example the graph (`reach`) is not displayed by default. If you have a look at the edited link of the codecov comment you can see the history of what the comment looked like. I ran it quite a few times to get it to contain the sections I wanted.

I can take another look to see if there are additional sections we can omit but I tried to copy the defaults from the WooCommerce core `yml` file.

**Comment by mikkamp** [comment]
I went ahead and removed more of those defaults as you suggested. I still left the comment.layout since I wasn't getting the `reach` section included initially. The `files` section is still not included, but that relies on comparing with a parent branch, so will only start showing up after we've merged to develop. It's ready for another round of review.

**Review by puntope — APPROVED** [review]
✅  LGTM

**Pull request merged**